Pd 400 cli guide document account group namespace (#1996)
truenas/documentation: content/SCALE/SCALECLIReference/Account/CLIgroup.md

{{< toc >}}

{{< include file="/_includes/CLI/CLIGuideWIP.md" type="page" >}}

> The **group** namespace contains 8 commands and is based on functions found in the SCALE API and web UI. It provides access to group account creation, configuration, and management.

Reviewer: numbers greater than 10 are typed as numbers, those lower are spelled out
```suggestion
The **group** namespace contains eight commands and is based on functions found in the SCALE API and web UI. It provides access to group account creation, configuration, and management.
```

You can enter commands from the main CLI prompt or from the system namespace prompt.

## Create Command
The `create` command creates a new group.

{{< include file="content/_includes/CLI/AccountGroupCreateExample.md" type="page" >}}

### Create Interactive Arguments Editor

The `create --` option opens an **interactive arguments editor**.

> {{< expand "Using the create Interactive Arguments Editor" "v" >}}

Reviewer: In this case this is a "heading" so use caps
```suggestion
{{< expand "Using the Create Interactive Arguments Editor" "v" >}}
```

> {{< trueimage src="/images/SCALE/CLI/Account/GroupCreateInteractiveArgumentsEditor.png" alt="Interactive Arguments Editor" id="1: Interactive Arguments Editor" >}}

Reviewer: is this the TUI for the create command? if yes, you can include that in the hover and caption text. I'm wondering if we should make this an expand so the reader can make the choice to see this content after the heading verses having it display by default. Thoughts?

Also, I think the content following this is better suited to that global content we talked about, then you can focus on the specifics of using the TUI for the create command in this article.

> {{< trueimage src="/images/SCALE/CLI/Account/GroupCreateInteractiveArgumentsEditor.png" alt="Interactive Arguments Editor" id="1: Interactive Arguments Editor" >}}

Author: ```suggestion
{{< trueimage src="/images/SCALE/CLI/Account/GroupCreateInteractiveArgumentsEditor.png" alt="Group Create Interactive Arguments Editor" id="1: Group Create Interactive Arguments Editor" >}}
```

In the TUI, most arguments are initially marked as comments with the `#` symbol, indicating that they are not yet configured. However, `name:` is shown as a required field.

To provide values for the other arguments, you need to remove the `#` comment designator from the corresponding line in the TUI.

{{< hint type=tip >}}
A group name is the only required value for group creation. If a group identification number is not provided, it is automatically filled with the next one available. All other arguments are optional.

See the relevant rows in [Configuration Arguments](#create-configuration-arguments) for more information.
{{< /hint >}}

A space is required between the provided argument and entered data, for example `name: test_group`.

Press <kbd>F2</kbd> or click **Save** to save the modified file.

Press <kbd>F10</kbd>, <kbd>Esc</kbd>, or click **Quit** to exit the TUI.
The `create` command automatically executes upon exit.
{{< /expand >}}

### Create Configuration Arguments

> `create` has 7 available arguments for group configuration. They are:

Reviewer: ```suggestion
`create` has seven available arguments for group configuration. They are:
```
I see the table below with the arguments and description details. Maybe remove this bullet list and tell them to see the table below.

Author: My thought was this gives simplified version so that the list is not completely buried in the expand, but then there is also more detail there

| `allow_duplicate_gid=` | Boolean | If set to true, allows distinct group names to share the same group identification number. Defaults to false. <br> Important: Use only if absolutely necessary. Duplicate GIDs can lead to unexpected behavior. <br> Ex. `allow_duplicate_gid=true` <br> &emsp; Where `true` is a boolean value. | No |
| `users=` | Array | Assigns users to the group with a list of user identification numbers (UIDs). <br> Ex. `users=[3001,3002]` <br> &emsp; Where `3001` and `3002` are UID numbers to assign to the group. | No |

{{< /truetable >}}
{{< /expand >}}

## Delete Command
The `delete` command deletes an existing group.

> `delete` has one command option.

Reviewer: What is the one option this command has? Include it here, and also in the command examples provided next. If it is required, then include it in the examples.

Author: It's in the expand snippet for now (I think it's to delete group members at the same time?) but this might get reworked when I get to that one.

{{< toc >}}

{{< include file="/_includes/CLI/CLIGuideWIP.md" type="page" >}}

The **group** namespace contains 8 commands and is based on functions found in the SCALE API and web UI. It provides access to group account creation, configuration, and management.

You can enter commands from the main CLI prompt or from the system namespace prompt.

> ### The create Command

Reviewer: we don't include "The" in the headings, suggested change: 
```suggestion
### Create Command
```

The `create` command creates a new group.

{{< expand "Using the create Command" "v" >}}

> From the CLI prompt, enter:

Reviewer: after the expand shortcode, provide more information/details about this command. the options, included, what is involved in using the command. We explain the command in more details. The reason we hide this behind the first expand is if we decide to drop this section into a non-CLI tutorial as a snippet (that includes the expands), that snippet has the information the reader needs in the tutorial.

`account group create`

From the **account** prompt, enter:

`group create`

{{< expand "Command Example" "v" >}}
{{< include file="content/_includes/CLI/AccountGroupCreateExample.md" type="page" >}}
{{< /expand >}}
{{< /expand >}}

`create` has one command option and 7 arguments.

### The create Interactive Arguments Editor

> Entering the `create --` option will open an **interactive arguments editor**.

Reviewer: never use "will" in the articles unless mentioning a future change to the feature, such as this command will be removed in a future release, suggestion for active voice present tense:
```suggestion
Entering the `create --` option opens an **interactive arguments editor**.
```
